src/gt4py/next/type_inference.py

```diff
@@ -142,11 +142,10 @@ def _free_variables(x: Type) -> set[TypeVar]:
 class _Dedup(eve.NodeTranslator):
     """Deduplicate type nodes that have the same value but a different `id`."""
 
-    def visit(self, node, *, memo: dict[T, T]) -> typing.Any:  # type: ignore[override]
+    def visit(self, node: Type | typing.Sequence[Type], *, memo: dict[Type, Type]) -> typing.Any:  # type: ignore[override]
```

Comment:
`typing.Sequence` is deprecated and should be replace by `collections.abc.Sequence`